server/sonar-server/src/main/java/org/sonar/server/batch/ProjectRepositoryLoader.java

```diff
@@ -88,8 +89,9 @@ public ProjectRepositories load(ProjectRepositoryQuery query) {
       if (module != null) {
         if (query.isPreview()) {
           // Scan permission is enough to analyze all projects but preview permission is limited to projects user can access
-          UserSession.get().checkComponentPermission(UserRole.USER, query.getModuleKey(),
-            "You're not authorized to access to project '" + module.name() + "', please contact your SonarQube administrator.");
+          if (!UserSession.get().hasProjectPermissionByUuid(UserRole.USER, module.projectUuid())) {
```

Comment:
@jblievremont Indeed, checkComponentPermission will first search the project of the component, then check permission on the found project.
But direct check on hasProjectPermissionByUuid or checkProjectUuidPermission will only check permission on the project.

To sum up, there's one less SQL query when using XXXProjectUuisPermission !